src/utils/getUserDetails.ts

```diff
@@ -0,0 +1,18 @@
+import { RDS_BASE_API_URL } from "../constants/urls";
+import { UserResponseType } from "../typeDefinitions/rdsUser.types";
+
+async function getUserDetails(id: string): Promise<UserResponseType> {
+  try {
+    const response = await fetch(
+      `${RDS_BASE_API_URL}/users?discordId=${id}&dev=true`
+    );
+
+    const userResponse = (await response.json()) as UserResponseType;
+    return userResponse;
+  } catch (error) {
+    console.error("An error occurred:", error);
+    throw error;
```

Comment:
you can remove the throw statement from the catch block and instead return a rejected promise. This will allow the caller to handle the error appropriately.